src/axom/primal/tests/primal_bezier_curve.cpp

```diff
@@ -142,6 +142,37 @@ TEST(primal_beziercurve, evaluate)
   }
 }
 
+//------------------------------------------------------------------------------
+TEST(primal_beziercurve_, point_derivatives)
+{
+  SLIC_INFO("Testing Bezier derivative calculations on a low-order curve");
+
+  const int DIM = 3;
+  using CoordType = double;
+  using PointType = primal::Point<CoordType, DIM>;
+  using VectorType = primal::Vector<CoordType, DIM>;
+  using BezierCurveType = primal::BezierCurve<CoordType, DIM>;
+
+  PointType data1[1] = {PointType {0.6, 1.2, 1.0}};
+  PointType data2[2] = {PointType {0.6, 1.2, 1.0}, PointType {1.3, 1.6, 1.8}};
+
+  BezierCurveType pointCurve(data1, 0);
+  BezierCurveType lineCurve(data2, 1);
+
+  // Evaluate the curve at several parameter values
+  // Curve should be tangent to control net at endpoints
+  VectorType eval_point_first = pointCurve.dt(0.5);
+  VectorType eval_point_second = pointCurve.dtdt(0.5);
+  VectorType eval_line_second = lineCurve.dtdt(0.5);
```

Comment:
Minor: Would it be worthwhile checking at other parameters, e.g. `t==0` and `t==1`?

Reply:
Yes, it would. I also think there could be some value in adding early returns to `BezierCurve.evaluate(t=0.0/1.0)` and `BezierPatch(u=0.0/1.0, v=0.0/1.0)` for those specific interpolating points, since it feels more natural (to me at least) to get those points through `.evaluate()`, even if `BezierCurve[0/ord]` and `BezierPatch(0/ord_u, 0/ord_v)` is the faster access.